lighthouse-cli/bin.ts

```diff
@@ -229,11 +235,32 @@ function lighthouseRun(addresses: Array<string>, config: Object, flags: Object)
   return lighthouse(address, flags, config)
     .then((results: Results) => Printer.write(results, outputMode, outputPath))
     .then((results: Results) => {
+      const filename = `${assetSaver.getFilenamePrefix({url: address})}.report.html`;
+
       if (outputMode === Printer.OutputMode[Printer.OutputMode.pretty]) {
-        const filename = `./${assetSaver.getFilenamePrefix({url: address})}.report.html`;
         Printer.write(results, 'html', filename);
       }
 
+      // Generate report.html, host it and open it in the default browser
+      if (flags.view) {
+        server.getPort(0).then((port: number) => {
+          const filePath = `${server.FOLDERS.REPORTS}/${filename}`;
+          const htmlExist = outputMode === Printer.OutputMode[Printer.OutputMode.pretty]
+            || (outputPath !== 'stdout' && outputMode === Printer.OutputMode[Printer.OutputMode.html]);
+
+          // Only generate HTML file when it is not already generated
+          // If the HTML file is already generated, create a symlink to it
+          if (htmlExist) {
+            const targetPath = outputMode === Printer.OutputMode[Printer.OutputMode.pretty] ? filename : outputPath;
+            fs.symlinkSync(path.resolve(targetPath), filePath);
```

Comment:
Instead of deleting the entire folder, now the server will delete all broken symlinks on start. So we can keep the old reports and we don't need to worry about those broken symlinks.